backend/code/m-booking/src/test/java/com/mbooking/service/ReservationServiceIntegrationTests.java

```diff
@@ -0,0 +1,370 @@
+		ReservationDTO reservationDTO = new ReservationDTO();
+		reservationDTO.setReservationDetails(new ArrayList<ReservationDetailsDTO>());
+		reservationDTO.setManifestationId(-2L);
+		for (int i = 0; i < 2; i++) {
+			ReservationDetailsDTO resdto = new ReservationDetailsDTO();
+			resdto.setColumn(0);
+			resdto.setRow(0);
+			resdto.setManifestationDayId(-3L);
+			resdto.setManifestationSectionId(-4L);
+			reservationDTO.getReservationDetails().add(resdto);
+		}
+		
+		SecurityContextHolder.getContext().setAuthentication(this.authentication);
+		reservationService.makeReservation(reservationDTO);
+	}
+	
+	@Test(expected = SeatTakenException.class)
+	@Rollback
+	@Transactional
+	public void test_makeReservation_SeatTaken() {
+		SecurityContextHolder.getContext().setAuthentication(this.authentication);
+		reservationService.makeReservation(reservationDTO);
+		reservationService.makeReservation(reservationDTO);
+	}
+	
+	@Test(expected = NoSuchSeatException.class)
+	@Rollback
+	@Transactional
+	public void test_makeReservation_NoSuchSeat_Negative() {
+		ReservationDetailsDTO reservationDetailsDTO1 = new ReservationDetailsDTO();
+		reservationDetailsDTO1.setColumn(-34);
+		reservationDetailsDTO1.setRow(-34);
+		reservationDetailsDTO1.setManifestationDayId(-1L);
+		reservationDetailsDTO1.setManifestationSectionId(-1L);
+		this.reservationDTO.getReservationDetails().add(reservationDetailsDTO1);
+		
+		SecurityContextHolder.getContext().setAuthentication(this.authentication);
+		reservationService.makeReservation(reservationDTO);
+	}
+	
+	@Test(expected = NoSuchSeatException.class)
+	@Rollback
+	@Transactional
+	public void test_makeReservation_NoSuchSeat_Positive() {
+		ReservationDetailsDTO reservationDetailsDTO1 = new ReservationDetailsDTO();
+		reservationDetailsDTO1.setColumn(34);
+		reservationDetailsDTO1.setRow(34);
+		reservationDetailsDTO1.setManifestationDayId(-1L);
+		reservationDetailsDTO1.setManifestationSectionId(-1L);
+		this.reservationDTO.getReservationDetails().add(reservationDetailsDTO1);
+		
+		SecurityContextHolder.getContext().setAuthentication(this.authentication);
+		reservationService.makeReservation(reservationDTO);
+	}
+	
+	@Test(expected = NoSuchManifestationDayException.class)
+	@Rollback
+	@Transactional
+	public void test_makeReservation_NoSuchManifestationDay() {
+		ReservationDetailsDTO reservationDetailsDTO1 = new ReservationDetailsDTO();
+		reservationDetailsDTO1.setColumn(1);
+		reservationDetailsDTO1.setRow(1);
+		reservationDetailsDTO1.setManifestationDayId(-3L);
+		reservationDetailsDTO1.setManifestationSectionId(-1L);
+		this.reservationDTO.getReservationDetails().add(reservationDetailsDTO1);
+		
+		SecurityContextHolder.getContext().setAuthentication(this.authentication);
+		reservationService.makeReservation(reservationDTO);
+	}
+	
+	@Test(expected = MaxReservationsException.class)
+	@Rollback
+	@Transactional
+	public void test_makeReservation_MaxReservations_Total() {
```

Comment:
givenTooManyReservations_whenMakingReservation_throwException ? 
What is total?